infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLog.java

```diff
@@ -137,15 +142,41 @@ default OverwriteCause getOverwriteCauseForAddress(long address, LogData entry)
     /**
      * Query if the StreamLog has enough quota to accept writes
      */
-    default boolean quotaExceeded() {
-        return false;
-    }
+    boolean quotaExceeded();
 
     /**
      * Query the exact Quota value in bytes
      * @return the quota set in bytes
      */
-    default long quotaLimitInBytes() {
-        return Long.MAX_VALUE;
-    }
+    long quotaLimitInBytes();
+
+    /**
+     * Query the space used in bytes
+     * @return the space used in bytes
+     */
+    ResourceQuota getLogSizeQuota();
+
+    /**
+     *
+     * @param startAddress the startAddress of a segment file
```

Comment:
Can you add some documentation here.

Reply:
+1 it is not clear why the endAddress is missing.